components/org.wso2.identity.apps.taglibs.layout.controller/src/main/java/org/wso2/identity/apps/taglibs/layout/controller/compiler/parsers/DefaultParser.java

```diff
@@ -0,0 +1,228 @@
+                        context.textKeeper = null;
+                    } else {
+                        allIdentifiers.add(
+                                new NoIdentifier(context.line.substring(context.start, context.matcher.start())));
+                    }
+                    context.start = context.matcher.end();
+                    return createCompiledObject(allIdentifiers);
+                } else if ((currentIdentifierName = context.matcher.group(5)) != null) {
+                    NotConditionIdentifier notCondition;
+                    if (context.textKeeper != null) {
+                        notCondition = new NotConditionIdentifier(currentIdentifierName,
+                                context.textKeeper + context.line.substring(context.start, context.matcher.start()));
+                        context.textKeeper = null;
+                    } else {
+                        notCondition = new NotConditionIdentifier(currentIdentifierName,
+                                context.line.substring(context.start, context.matcher.start()));
+                    }
+                    context.start = context.matcher.end();
+                    ExecutableIdentifier child = compile(reader, context, currentIdentifierName);
+                    notCondition.setChild(child);
+                    allIdentifiers.add(notCondition);
+                }
+            } else {
+                if (context.textKeeper == null) {
+                    context.textKeeper = new StringBuilder(context.line.substring(context.start) + "\n");
+                } else {
+                    context.textKeeper.append(context.line.substring(context.start) + "\n");
+                }
+                context.line = "";
+                context.matcher = null;
+                context.start = 0;
+            }
+        }
+        if (identifierName != null) {
+            throw new IllegalStateException(
+                    "Layout file is not correctly written, make sure to place closing tags in correct place");
+        }
+        if (context.textKeeper != null) {
+            allIdentifiers.add(new NoIdentifier(context.textKeeper.toString()));
+        }
+        return createCompiledObject(allIdentifiers);
+    }
+
+    /**
+     * Create the final compiled version of the layout file.
+     *
+     * @param allIdentifiers Array of identifiers.
+     * @return All identifiers as a single object.
+     */
+    private ExecutableIdentifier createCompiledObject(ArrayList<ExecutableIdentifier> allIdentifiers) {
+
+        return new DefaultIdentifier(allIdentifiers);
+    }
+
+    /**
+     * Read a new line from the layout file.
+     *
+     * @param reader  BufferedReader to read the layout file.
+     * @param context Compiling context.
+     * @return Whether is there new line to read.
+     */
+    private boolean readLine(BufferedReader reader, CompileContext context) {
+
+        if (context.line.equals("")) {
```

Comment:
```suggestion
        if (StringUtils.EMPTY.equals(context.line)) {
```
Also do null check for context prior to this if check to avoid possible NPE